batchproducer/batchproducer.go

```diff
@@ -233,15 +232,22 @@ func (b *batchProducer) Stop() error {
 }
 
 func (b *batchProducer) setRunning(running bool) {
-	b.runningMu.Lock()
-	defer b.runningMu.Unlock()
-	b.running = running
+	var newValue int32
+	if running {
+		newValue = 1
+	} else {
+		newValue = 0
+	}
+
+	oldValue := b.running
+
+	for swapped := false; !swapped; {
```

Comment:
I was getting data race warnings when I passed `-race` to `go test`. Since I’ll be contributing this back to the main repo, and I’m hoping they’ll merge it, I was thinking that eliminating data race warnings would be prudent.